tests/components/tado/test_config_flow.py

```diff
@@ -22,7 +27,73 @@ def _get_mock_tado_api(getMe=None):
     return mock_tado
 
 
-async def test_form(hass: HomeAssistant) -> None:
+@pytest.mark.parametrize(
+    ("exception", "error"),
+    [
+        (KeyError, "invalid_auth"),
+        (RuntimeError, "cannot_connect"),
+        (ValueError, "unknown"),
+    ],
+)
+async def test_form_exceptions(
```

Comment:
Just reconfirmed this with another core member, and we still require config flow exception tests to make sure the config flow is able to recover from a failure. (For example, not that if a timeout occurs, all next tries will fail because of that first failure).

The full flow test logic can be reused here.

Reply:
Thanks for the guidance and feedback. Is this also written down somewhere or examplified? I am looking down the documentation, but I seem to be missing requirements of the standard of unit tests (other than that the config flow should be 100% covered).

Reply:
Good point, I'll try to get this cleared up in the docs. 

I have a recent example in a PR I made myself #98220. In `test_config_flow.py` you can see that it retries the flow with a valid mock and then we check if the config entry is made as expected.

Reply:
I'll look into it, thanks!

Reply:
To make a better example, line 105 to line 126 of the Tado config flow test contains the logic needed to complete the flow

Reply:
Thanks for clarifying! I think I understand your point. A successful retry/creation has been added at the bottom of the exceptions. Let me know if this is what is desired and standard. :)